File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    dic = {}
    result = []
    for i in range(len(name)):
        dic[name[i]] = yearning[i]
    
    for p in photo:
        tmp = 0
        for t in p:
            if t in dic:
                tmp += dic[t]
        
        result.append(tmp)
        
    return result
            
        
        

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        for (int i=0; i<photo.length; i++) {
            int sum = 0;
            for (int j=0; j<photo[i].length; j++) {
                for (int k=0; k<name.length; k++) {
                    if (name[k].equals(photo[i][j])) {
                        sum += yearning[k];
                    }
                }
            }
            answer[i] = sum;
        }
        
        return answer;
    }
}